internal/storagetools/delete_object_handler.go

```diff
@@ -0,0 +1,17 @@
+package storagetools
+
+import (
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/pkg/storages/storage"
+)
+
+func HandleDeleteObject(objectPath string, folder storage.Folder) {
+	// some storages may not produce an error on deleting the non-existing object
+	exists, err := folder.Exists(objectPath)
+	tracelog.ErrorLogger.FatalfOnError("Failed to check object existence: %v", err)
+	if !exists {
+		tracelog.ErrorLogger.Fatalf("Object %s does not exist", objectPath)
+	}
+	err = folder.DeleteObjects([]string{objectPath})
```

Comment:
Does every storage engine support compex object paths like `a/b/c/d.zip` ? 
Don't we need to traverse folder down to with SubFolder().SubFolder() before accessing object ?

Reply:
S3 should support this. As for other storages, I think that this logic needs to be implemented in the corresponding storage implementation. I propose to leave it like this here.